receiver/mysqlreceiver/factory.go

```diff
@@ -0,0 +1,66 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysqlreceiver
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+)
+
+const (
+	typeStr = "mysql"
+)
+
+func NewFactory() component.ReceiverFactory {
+	return receiverhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiverhelper.WithMetrics(createMetricsReceiver))
+}
+
+func createDefaultConfig() config.Receiver {
+	return &Config{
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			ReceiverSettings:   config.NewReceiverSettings(config.NewID(typeStr)),
+			CollectionInterval: 10 * time.Second,
+		},
+		Endpoint: "localhost:3306",
+	}
+}
+
+func createMetricsReceiver(
+	_ context.Context,
+	params component.ReceiverCreateSettings,
+	rConf config.Receiver,
+	consumer consumer.Metrics,
+) (component.MetricsReceiver, error) {
+	return nil, nil
+	// cfg := rConf.(*Config)
+
+	// ns := newMySQLScraper(params.Logger, cfg)
+	// scraper := scraperhelper.NewResourceMetricsScraper(cfg.ID(), ns.scrape, scraperhelper.WithStart(ns.start),
+	// 	scraperhelper.WithShutdown(ns.shutdown))
+
+	// return scraperhelper.NewScraperControllerReceiver(
+	// 	&cfg.ScraperControllerSettings, params.Logger, consumer,
+	// 	scraperhelper.AddScraper(scraper),
+	// )
```

Comment:
No commented code please :)

Reply:
Also, need to return receiver implementation here or it won't be registered correctly.

Reply:
Removed for this PR.

@MikeGoldsmith The implementation is in the next PR. This is decomposed per contribution guidelines.